src/HW/client/src/components/List/List.jsx

```diff
@@ -1,30 +1,23 @@
-import ListItem from "../ListItem/ListItem.jsx";
 import "./List.css";
 import PropTypes from "prop-types";
-const List = ({ items, deleteItemFromDb, updateStatusDb, editTaskNameDb }) => {
+import ListItemConnector from "../ListItem/ListItemConnector.js";
+const List = ({ items, searchInputValue}) => {
   return (
     <div>
       <ul className="tasks">
-        {items.map((item, index) => {
-          return (
-            <ListItem
-              item={item}
-              deleteItemFromDb={deleteItemFromDb}
-              updateStatusDb={updateStatusDb}
-              key={index}
-              editTaskNameDb={editTaskNameDb}
-            />
-          );
-        })}
+        
+        {items
+          .filter((item) => item.itemName.includes(searchInputValue))
```

Comment:
in order to avoid calculating it in every render , you can move it to the redus state and move this logic to a selector. when the search filter is change - dispatch an action that saves the new text, and the selector will calculate the tasks according to it